# 0 is upper
# 1 is rota

int8 client_num

---

bool success